src/main/cpp/src/map_zip_with_utils.cu

```diff
@@ -0,0 +1,308 @@
+  auto const labels_begin = labels->mutable_view().template begin<size_type>();
+  cudf::detail::label_segments(
+    input.offsets_begin(), input.offsets_end(), labels_begin, labels_begin + n_elements, stream);
+  return labels;
+}
+
+[[maybe_unused]] std::unique_ptr<column> indices_of(
+  lists_column_view const& search_keys,    // Column containing lists of keys to search with
+  lists_column_view const& search_values,  // Column containing lists of values to search for
+  rmm::cuda_stream_view stream,            // CUDA stream for asynchronous execution
+  rmm::device_async_resource_ref mr)       // Memory resource for allocations
+{
+  CUDF_EXPECTS(search_keys.size() == search_values.size(),
+               "Number of search keys lists must match search values lists.");
+
+  // Get the number of lists (rows) in the input columns
+  auto const num_lists = search_values.size();
+
+  // Extract the child columns containing the actual key and value data
+  auto const all_keys   = search_keys.child();
+  auto const num_keys   = all_keys.size();
+  auto const all_values = search_values.child();
+
+  // Calculate the number of elements in each list for both keys and values
+  auto list_sizes = cudf::lists::count_elements(search_values, stream, mr);
+  auto keys_sizes = cudf::lists::count_elements(search_keys, stream, mr);
+
+  // Generate labels to map each key back to its corresponding list index
+  // This helps us know which list each key belongs to
+  auto keys_labels = generate_labels(search_keys, num_keys, stream, mr);
+
+  // For each key, get the size of the associated values list
+  // This tells us how many values we need to compare against for each key
+  auto associated_list_sizes = thrust::make_transform_iterator(
+    keys_labels->view().begin<size_type>(),
+    [list_sizes_ptr = list_sizes->view().begin<size_type>(), num_lists] __device__(
+      auto const offset_val) { return offset_val >= num_lists ? 0 : list_sizes_ptr[offset_val]; });
+
+  // Calculate cumulative offsets for the associated list sizes
+  // This gives us the starting position for each key's value comparisons
+  auto list_sizes_offsets = make_numeric_column(
+    data_type(type_to_id<size_type>()), num_keys + 1, cudf::mask_state::UNALLOCATED, stream, mr);
+  auto d_list_sizes_offsets = list_sizes_offsets->mutable_view().template data<size_type>();
+  thrust::exclusive_scan(rmm::exec_policy(stream),
+                         associated_list_sizes,
+                         associated_list_sizes + num_keys + 1,
+                         d_list_sizes_offsets);
+
+  // Calculate the total number of comparisons needed for each row
+  // For each list, we need: number_of_keys * number_of_values comparisons
+  auto total_compares_per_row = thrust::make_transform_iterator(
+    thrust::make_counting_iterator(0),
+    [keys_sizes = keys_sizes->mutable_view().template data<size_type>(),
+     list_sizes = list_sizes->mutable_view().template data<size_type>(),
```

Comment:
Am I reading this right?  These shouldn't be `mutable_view`, right?  These are only inputs.